diff -urN a/arch/mips/boot/dts/ralink/Makefile b/arch/mips/boot/dts/ralink/Makefile
--- a/arch/mips/boot/dts/ralink/Makefile	2019-03-06 18:31:45.760354495 -0500
+++ b/arch/mips/boot/dts/ralink/Makefile	2019-03-06 18:32:28.480314335 -0500
@@ -5,6 +5,7 @@
 dtb-$(CONFIG_DTB_MT7620A_EVAL)	+= mt7620a_eval.dtb
 dtb-$(CONFIG_DTB_OMEGA2P)	+= omega2p.dtb
 dtb-$(CONFIG_DTB_VOCORE2)	+= vocore2.dtb
+dtb-y				+= LINKIT7688.dtb
 
 obj-y				+= $(patsubst %.dtb, %.dtb.o, $(dtb-y))
 
